### DISCUSSION TITLE
Spanning intervals over breaks

### DISCUSSION THREAD

**Discussion by andrea-buso** [google/or-tools#3813] [CP-SAT questions]
I'd like to generalize the code in
https://github.com/google/or-tools/blob/stable/ortools/sat/docs/scheduling.md#intervals-spanning-over-breaks-in-the-calendar
with the use of n intervals over more than one day.
The example above considers the breaks already known in advance: what if I have the breaks expressed with fixed intervals that may vary from one day to another?
Thanks
Andrea

**Comment by lperron** [comment]
The code does not depend on the interval being repeated every day. It depends on the intervals being fixed.

**Reply by andrea-buso** [reply]
Thanks Laurent for you help.
I see what you mean.
I followed an idea of mine and I tried to build a model where each original interval is split over the breaks, but the complexity tends to be very high, even for a small number of intervals/breaks
What strategy would you suggest with CP-SAT?
Andrea

**Comment by Hakunari** [comment]
Hello! Did you find any solution for this? I have the save problem.

**Reply by gregy4** [reply]
I think the example is little bit misleading.
I would start with channeling constraints and boolean variable for every possible value of start variable, calculation of duration is up to you. This of course can lead to a lot of channeling constraints and boolean variables. Since you can probably find that some neighbor positions lead to same duration, a value of start variable can be replaced by an interval. I'm not sure whether joining of intervals (in cp-sat scheduling recipes) to save boolean variables leads to better performance.
In some cases I use even multiple calendars with large datasets and so far I didn't noticed performance problems related to this part of modelling.
